test/ui/LoginTestPage.ts

```diff
@@ -0,0 +1,100 @@
+import { By, WebDriver } from 'selenium-webdriver';
+import { expect } from 'chai';
+import { driverBuilder } from './driverBuilder'; // Adjust the path as necessary
+
+export class LoginTestPage {
+    private driver: WebDriver;
+
+    static URL: string = 'http://localhost:3000/loginForm';
+
+    private usernameField: By = By.id('Username');
+    private passwordField: By = By.id('Password');
+    private loginButton: By = By.id('login');
+    private logoutButton: By = By.id('logoutButton');
+    private loggedToHomepageSuccessfully: By = By.id("home-page-title");
+    private loggedOutSuccessfully: By = By.id("loginFrom");
+    private errorMessage: By = By.id('error-message');
+
+    constructor(driver: WebDriver) {
+        this.driver = driver;
+    }
+
+    static async initialize(): Promise<LoginTestPage> {
+        const driver = await driverBuilder.driverBefore();
+        return new LoginTestPage(driver);
+    }
+    async open(url: string): Promise<void> {
+        await this.driver.get(url);
+    }
+
+    async enterUsername(username: string): Promise<void> {
```

Comment:
Do all of these tests need to split into seperate tests just for entering data? Can you merge them together to create one login test

Reply:
Also does this not do the same as testLoginSuccess?